public/app/features/dashboard/components/DashExportModal/DashboardExporter.ts

```diff
@@ -132,16 +139,29 @@ export class DashboardExporter {
             return;
           }
 
-          const refName = 'DS_' + ds.name.replace(' ', '_').toUpperCase();
+          const libraryPanel = obj.libraryPanel;
+          const libraryPanelSuffix = !!libraryPanel ? '-for-library-panel' : '';
+          let refName = 'DS_' + ds.name.replace(' ', '_').toUpperCase() + libraryPanelSuffix.toUpperCase();
+
           datasources[refName] = {
             name: refName,
             label: ds.name,
             description: '',
             type: 'datasource',
             pluginId: ds.meta?.id,
             pluginName: ds.meta?.name,
+            usage: datasources[refName]?.usage,
           };
 
+          if (!!libraryPanel) {
+            const libPanels = datasources[refName]?.usage?.libraryPanels || [];
+            libPanels.push({ name: libraryPanel.name, uid: libraryPanel.uid });
+
+            datasources[refName].usage = {
+              libraryPanels: libPanels,
+            };
+          }
+
```

Comment:
I add an array of library panels inside usage in order to know which library panels are using that datasource.